src/field_theory/is_alg_closed/basic.lean

```diff
@@ -158,13 +158,14 @@ algebra_map_surjective_of_is_integral ((is_algebraic_iff_is_integral' k).mp hf)
 end is_alg_closed
 
 /-- Typeclass for an extension being an algebraic closure. -/
-class is_alg_closure (K : Type v) [field K] [algebra k K] : Prop :=
+class is_alg_closure (R : Type u) (K : Type v) [comm_ring R] [field K] [algebra R K] : Prop :=
 (alg_closed : is_alg_closed K)
-(algebraic : algebra.is_algebraic k K)
+(algebraic : algebra.is_algebraic R K)
+(injective : function.injective (algebra_map R K))
```

Comment:
it may be worth making this a TC argument with `no_zero_smul_divisors R K` instead an injective `algebra_map`. (they're equivalent)